daffodil-runtime2/src/main/scala/org/apache/daffodil/runtime2/generators/CodeGeneratorState.scala

```diff
@@ -217,10 +232,12 @@ class CodeGeneratorState {
         case PrimType.Int => "int32_t    "
         case PrimType.Short => "int16_t    "
         case PrimType.Byte => "int8_t     "
+        case PrimType.Float => "float      "
+        case PrimType.Double => "double     "
         case x => context.SDE("Unsupported primitive type: " + x)
       }
     } else {
-      child.namedQName.local + "         "
+      localName(child)
```

Comment:
Feels like you want qualfiedName(child) here, but you want to convert that into something you can use as a field name.

At some point we'll have to pick a preferred prefix for a namespace from the available prefixes for a namespace, and then construct a field name by prefix_localName or some such formula.

Reply:
Hmm, I didn't think we would need to qualify C struct field names since their names only need to be unique inside each C struct's scope.  Are you thinking somebody might define a complex element with two children (either simple or complex elements) having the same local name in two different namespaces too?  We would need a prefix_localName in that case too.